packages/cross_file/lib/src/types/io.dart

```diff
@@ -82,9 +92,11 @@ class XFile extends XFileBase {
   @override
   String get path => _file.path;
 
+  /// The name extracted from the last part of [File.path].
   @override
   String get name => _file.path.split(Platform.pathSeparator).last;
 
+  /// The [File.length].
   @override
   Future<int> length() {
```

Comment:
`name` and `length` are documented in the base class.

The documentation of the base class for [`name`](https://github.com/flutter/packages/blob/51e74b97508a3f85af2c45d994be0e0a58b496f8/packages/cross_file/lib/src/types/base.dart#L39-L44) and [`length`](https://github.com/flutter/packages/blob/main/packages/cross_file/lib/src/types/base.dart#L51-L54) may be a better place to fix?

Reply:
Yes it could work too. I placed that there because it make more sense to me to have specificity of an implementation declared as close as possible to the actual class instead of to the abstract class. Anyway, I'm seeing right now that in the base class you are already documenting the return value of some methods based on the implementation it may be. I'll try to do the same for these ones.

Reply:
Thanks! I'm not exactly sure what happens to the [generated documentation](https://pub.dev/documentation/cross_file/0.3.3+5/cross_file/XFile/name.html) when a method with `@override` has its own docs (not sure if they get merged, or the base class one is ignored!)